src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallGlobalOrToolPathCommand.cs

```diff
@@ -93,9 +105,11 @@ public override int Execute()
                 toolPath = new DirectoryPath(_toolPath);
             }
 
+            string appHostSourceDirectory = ResolveAppHostSourceDirectoryAsync(_architectureOption, _framework).Result;
+
             (IToolPackageStore toolPackageStore, IToolPackageStoreQuery toolPackageStoreQuery, IToolPackageInstaller toolPackageInstaller) =
                 _createToolPackageStoresAndInstaller(toolPath, _forwardRestoreArguments);
-            IShellShimRepository shellShimRepository = _createShellShimRepository(toolPath);
+            IShellShimRepository shellShimRepository = _createShellShimRepository(toolPath, appHostSourceDirectory);
```

Comment:
the abstraction may not be on the right level. 

We also have the following:
shellShimRepository.CreateShim(command.Executable, command.Name, package.PackagedShims);